src/main/java/jp/ikedam/jenkins/plugins/extensible_choice_parameter/TextareaChoiceListProvider.java

```diff
@@ -172,8 +189,15 @@ protected void addEditedValue(
     {
         LOGGER.info(String.format("Add new value %s to parameter %s in project %s", value, def.getName(), project.getName()));
         List<String> newChoiceList = new ArrayList<String>(getChoiceList());
-        newChoiceList.add(value);
+        
+        // check if the new value should be added to the top of the list, or not
+        if (this.addToTop)
+        	newChoiceList.add(0, value);
+        else
+        	newChoiceList.add(value);
+

```

Comment:
1. There're tab characters. Please use white spaces for indentations.
2. I highly recommend you to use braces ({}) even for single statements. Blocks without braces often causes bugs especially when other developer add changes to your code.